eng/versioning.targets

```diff
@@ -22,8 +22,8 @@
     </AssemblyMetadata>
   </ItemGroup>
 
-  <!-- Adds SupportedOSPlatform attribute for Windows Specific libraries -->
-  <ItemGroup Condition="'$(IsWindowsSpecific)' == 'true' and '$(IsTestProject)' != 'true'">
+  <!-- Adds SupportedOSPlatform attribute for Windows Specific libraries and Windows targets -->
+  <ItemGroup Condition="('$(IsWindowsSpecific)' == 'true' or '$(TargetsWindows)' == 'true') and '$(IsTestProject)' != 'true'">
```

Comment:
Just for my purposes to better understand this. What's the difference between `IsWindowsSpecific` and `TargetsWindows`?

Reply:
IsWindowsSpecific was a hook added so that projects that are only supported on windows add that property and get the SupportedPlatform attribute into the assembly. I think with this change we might no longer need that property as I would expect those to have a Windows configuration.

Reply:
> I think with this change we might no longer need that property as I would expect those to have a Windows configuration.

That idea came up in the above comments, and I tried it, when using only `TargetsWindows` the Supported attribute is not being generated for ref assemblies even for windows only assemblies, so kept it as is

https://github.com/dotnet/runtime/pull/43097/files#r502720394

Reply:
Ah, right. That is because the ref assemblies never cross compile per runtime, so `TargetsWindows` will not be defined for them.